crates/attestation/src/attestation.rs

```diff
@@ -39,6 +45,55 @@ pub struct DstackAttestation {
     pub tcb_info: TcbInfo,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, thiserror::Error)]
+pub enum VerificationError {
+    #[error("could not parse embedded measurements: {0}")]
+    EmbeddedMeasurementsParsing(MeasurementsError),
+    #[error("dcap verification failed: {0}")]
+    DcapVerification(String),
+    #[error("verification report is not TD10")]
+    ReportNotTd10,
+    #[error("TCB status `{0}` is not up to date")]
+    TcbStatusNotUpToDate(String),
+    #[error("ouststanding advisories reported: {0}")]
+    NonEmptyAdvisoryIds(String),
+    #[error("wrong {name} hash (found {found} expected {expected})")]
+    WrongHash {
+        name: &'static str,
+        found: String,
+        expected: String,
+    },
+    #[error("invalid event type {0}")]
+    InvalidEventType(u32),
+    #[error("failed to decode event digest `{0}`")]
+    EventDecoding(String),
+    #[error("failed to parse app compose JSON: {0}")]
+    AppComposeParsing(String),
+    #[error("no {0} event in event log")]
+    MissingEvent(&'static str),
+    #[error("duplicate {0} events in event log")]
+    DuplicateEvent(&'static str),
```

Comment:
question: why choosing `&'static str` here instead of `String` like in many other cases?

Reply:
Because these events are statically known and expected to be set when calling, whereas I've used `Strings` for nested error messages in other places. And because &'static str is more conservative than String, it's better to start with that imo.